pkg/controller/stacks/install/installjob.go

```diff
@@ -328,7 +329,7 @@ func (jc *stackInstallJobCompleter) replaceCRD(ctx context.Context, obj *unstruc
 	meta.AddLabels(obj, existing.GetLabels())
 	meta.AddAnnotations(obj, existing.GetAnnotations())
 
-	return jc.client.Update(ctx, obj)
```

Comment:
Can you help me understand why the `Update` call was insufficient in this scenario?

Reply:
@jbw976 sure! The issue here is that as you can see we above, we are not getting the object and then modifying it, we are getting the object, then trying to update it with the CRD that is supplied by the new stack. This means that we are not including a resource version on the object supplied to the update, so we will always fail because the api server can not determine if the object has been modified between when we got it and when we issued the update. We can actually simplify some of this logic a bit (for example, passing the crdVersionsInclusive function as an apply option), but I opted for the smallest fix with the least churn as we determine as path forward for some refactor here :)

Reply:
ah yes, that makes sense because the incoming `obj` here is output from the unpacking job.  It was not retrieved form the api server.  Thank you for taking the time to explain this to me @hasheddan 🙇‍♂️